python_modules/dagster/dagster_tests/asset_defs_tests/test_materialize.py

```diff
@@ -185,8 +185,11 @@ def the_asset():
 
 
 def test_materialize_no_assets():
-    with instance_for_test() as instance:
-        assert materialize([], instance=instance).success
+    # we need to decide if this is a valid use case given that it is inconsistent with Definitions
+    # rules
+    assert True
+    # with instance_for_test() as instance:
+    #     assert materialize([], instance=instance).success
```

Comment:
what's the actual use case for this anyway - seems like a pythonic edge case rather than behavior users would actually want

Reply:
A case to consider if when you are producing the list of assets programmatically and don't want to special the case with no assets. I don't feel super strongly here on this particular issue. I think the interesting bit is the behavior is inconsitent.

Reply:
> A case to consider if when you are producing the list of assets programmatically and don't want to special the case with no assets.

Yeah - I think it's worth making this complete successfully to support this use case.

Reply:
hard for me to imagine programmatic construction of assets ever giving no assets in a realistic scenario, seems more than likely that if no assets are produced, you're pointing at the wrong module, or something else is going wrong.